elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/models/Dimension.java

```diff
@@ -7,15 +7,44 @@
 
 import com.yahoo.elide.annotation.Include;
 import com.yahoo.elide.core.dictionary.EntityDictionary;
+import com.yahoo.elide.core.type.Type;
+import com.yahoo.elide.datastores.aggregation.annotation.ColumnMeta;
+import com.yahoo.elide.datastores.aggregation.annotation.DimensionFormula;
+
 import lombok.EqualsAndHashCode;
 
+import java.util.HashSet;
+
 /**
  * Regular field in tables, can be grouped by.
  */
 @Include(rootLevel = false, type = "dimension")
 @EqualsAndHashCode(callSuper = true)
 public class Dimension extends Column {
     public Dimension(Table table, String fieldName, EntityDictionary dictionary) {
-        super(table, fieldName, dictionary);
+        super(table, fieldName, dictionary, getConstructFunction(table, fieldName, dictionary));
+    }
+
+    protected static ConstructFunction getConstructFunction(Table table, String fieldName,
```

Comment:
I don't see a difference here between this construction mechanism and the one in Metric or TimeDimension.  Maybe we can remove this responsibility from these different classes and just construct this in Column?